direct/core/src/main/java/cz/o2/proxima/direct/commitlog/CommitLogReaders.java

```diff
@@ -129,17 +129,17 @@ public boolean hasExternalizableOffsets() {
   public static class LimitedCommitLogReader extends ForwardingCommitLogReader {
 
     @Getter private final ThroughputLimiter limiter;
-    private final Collection<Partition> partitions;
 
     public LimitedCommitLogReader(CommitLogReader delegate, ThroughputLimiter limiter) {
       super(delegate);
       this.limiter = SerializableUtils.clone(Objects.requireNonNull(limiter));
-      this.partitions = new ArrayList<>(delegate.getPartitions());
     }
 
     @Override
     public ObserveHandle observe(String name, Position position, LogObserver observer) {
-      return super.observe(name, position, throughputLimited(limiter, partitions, observer));
+      final List<Partition> availablePartitions = getDelegate().getPartitions();
```

Comment:
Can we cache those? It might be possible to call `observe` mutliple times for the same reader (e.g. after checkpoint in SDF), it would be good not to resolve partitions all over again.

Reply:
👍 done